### PR TITLE
slightly cleaner GetOrigin implementation

### PR DESCRIPTION
**Pull request by smcl**
I noticed that `StringExtensions.GetOrigin` is a little hard to follow, and uses `String.IndexOf` and substring to approximate URL parsing. While it works alright, I think it's probably better to use the `Uri` class.

### CODE REVIEW

**Comment by leastprivilege** [comment]
Thanks!

Coud you please add some tests to it?

**Comment by smcl** [comment]
Good call - I have a set of addresses I tested this against already (taking the output of existing implementation vs mine) but I'm not sure what the best way to add these are. Perhaps I should add a separate pull req with the tests (so we have a set of inputs we know pass against the existing implementation) - then it's possible to test for regressions in the new implementation? What do you think? I know it's not a hugely important part of IdentityServer, but I'd like to get it right :-)

**Comment by leastprivilege** [comment]
Either a new PR or add a commit. thanks

**Comment by brockallen** [comment]
Last chance to add tests. Thanks.

**Comment by smcl** [comment]
Hey Brock - sorry I have a lot of very poor excuses for not responding! I've added the tests that I put together - testing various combinations of scheme * host * port * resource against the value returned by the original implementation.

**Comment by brockallen** [comment]
Cool, thanks!
